### ISSUE TITLE
Support for specifying attributes inside project and model files

### ISSUE THREAD

**Issue by Kampfkarren** [rojo-rbx/rojo#262]
Once attributes come out, it would be awesome for Rojo to support them.

I'm interested in contributing this, if there's no more major refactors the codebase is going under.

**Comment by LPGhatguy** [comment]
The codebase is pretty stable now!

It might be easier to try adding support for tags first, which are available now. We'll want to reason through whether Rojo should merge or replace tag lists (merging might make us unable to _remove_ tags?) and some other edge cases we haven't hit before.

**Comment by Kampfkarren** [comment]
@LPGhatguy Would that require letting rbx_dom_weak reason about tags? No Rust code in that codebase references Tags at all, wasn't sure if that's an intentional decision or if it's just something no one got around to.

**Comment by LPGhatguy** [comment]
I don't think rbx_dom_weak should reason about types beyond how it does right now unless we have compelling reasons since that's a deep rabbit hole (terrain, meshes, collision data...)

We can implement this just by supporting the attribute format in Rojo, or even making a crate for it?

**Comment by Kampfkarren** [comment]
Making a crate for this kind of stuff (attributes *and* tags?) sounds fine to me, especially since it means that other developers could do things like reading attributes/tags in programs not related to Rojo.

**Comment by Ozzypig** [comment]
Hello from the future 👋

Attributes have been available for a few months now ([DevHub](https://developer.roblox.com/en-us/articles/instance-attributes), [Feb 23rd Announcement](https://devforum.roblox.com/t/attributes-now-available/1066934)). Are there any plans to update Rojo to support this fantastic new feature?

**Comment by Kampfkarren** [comment]
Yes, though it's waiting on this PR:

https://github.com/rojo-rbx/rbx-dom/pull/166

**Comment by Hexcede** [comment]
> Yes, though it's waiting on this PR:
> 
> [rojo-rbx/rbx-dom#166](https://github.com/rojo-rbx/rbx-dom/pull/166)

Someone should definitely work on this now that the PR has been merged into `rbx-dom`.
Would love to finally get to make use of attributes outside of my lua files. :smile:

(I would make a PR myself if I was comfortable with Rust but I'm not confident I wouldn't just make a buggy mess)

**Comment by Ozzypig** [comment]
Boy how time flies - any updates on Attribute support in Rojo?

If I were good at Rust I'd try my hand at this myself, but alas there's so many hours in a day 😞

**Comment by PoolloverNathan** [comment]
I will try my hand, I guess. Can't be that hard, right? (Famous last words)

**Comment by PoolloverNathan** [comment]
My plan is to have a `$attributes`/`Attributes` property, which can either be a simple type or an explicit cast.
